### PR TITLE
Feat/fused flashattention with packing

### PR DESCRIPTION
**Pull request by aolemila**
## Command

```bash
# start server
JAX_COMPILATION_CACHE_DIR=/tmp/jit_cache  python3 -u -m sgl_jax.launch_server --model-path Qwen/Qwen3-8B --trust-remote-code  --dist-init-addr=0.0.0.0:10011 --nnodes=1  --tp-size=4 --device=tpu --random-seed=3 --node-rank=0 --mem-fraction-static=0.8 --chunked-prefill-size=2048 --download-dir=/tmp --dtype=bfloat16 --max-running-requests 256 --skip-server-warmup --page-size=128  --disable-radix-cache

# eval(evalscope==0.17.1)
evalscope eval  --model Qwen-8B --api-url http://127.0.0.1:30000/v1/chat/completions --api-key EMPTY --eval-type service --datasets gsm8k --eval-batch-size 64

# benchmark
python3 -m sgl_jax.bench_serving --backend sgl-jax --dataset-name random --num-prompts 16 --random-input 1024 --random-output 1024 --random-range-ratio 1 --warmup-requests 0 --max-concurrency=48
```

## Accuracy

```bash
+---------+-----------+-----------------+----------+-------+---------+---------+
| Model   | Dataset   | Metric          | Subset   |   Num |   Score | Cat.0   |
+=========+===========+=================+==========+=======+=========+=========+
| Qwen-8B | gsm8k     | AverageAccuracy | main     |  1319 |  0.9098 | default |
+---------+-----------+-----------------+----------+-------+---------+---------+
```

### CODE REVIEW

File: python/sgl_jax/test/test_flashattention.py [hunk 1]
```diff
@@ -372,6 +372,7 @@ def run_test(self, mode, lens, mode_args):
             forward_batch.seq_lens,
             page_table,
             forward_batch.attn_backend.forward_metadata.cu_q_lens,
+            # forward_batch.attn_backend.forward_metadata.cu_kv_lens,
```
Comment (JamesBrianD) [inline python/sgl_jax/test/test_flashattention.py:375]: maybe you can remove this line
